Add support to parse multipart response body

dnlserrano opened the pull request:
Fixes #346

edgurgel commented:
Hey @dnlserrano, 

I was thinking about this solution and my only with it is that we would need to parse every header to find out if we should decode or not. I wonder if we could provide a "helper function" that would get the final `HTTPoison.Response` struct and decode if multipart? So if users have endpoints that will return this kind of body response they can use it etc?

What do you think?

dnlserrano commented:
That all sounds great! I've updated the PR with your proposed changes. Does the module `HTTPoison.Handlers.Multipart` make sense?

I've also added hexdocs and READ documentation for multipart. Let me know what you think! 👊

edgurgel commented:
This PR is awesome <3

Amazing really 🎉 

I will release a new version soon 👌
